src/VirtoCommerce.Platform.Web/Telemetry/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using VirtoCommerce.Platform.Core.Telemetry;
+
+namespace VirtoCommerce.Platform.Web.Telemetry
+{
+    public static class ApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Configure AppInsights sampling accordingly to
+        /// https://docs.microsoft.com/en-us/azure/azure-monitor/app/sampling#configure-sampling-settings
+        /// </summary>
+        /// <param name="app"></param>
+        /// <returns></returns>
+        public static IApplicationBuilder UseAppInsightsTelemetry(this IApplicationBuilder app)
+        {
+            var samplingOptions = app.ApplicationServices.GetRequiredService<IOptions<SamplingOptions>>().Value;
+
+            var configuration = app.ApplicationServices.GetService<TelemetryConfiguration>();
+
+            var builder = configuration.DefaultTelemetrySink.TelemetryProcessorChainBuilder;
+            if (samplingOptions.Processor == SamplingProcessor.Adaptive)
+            {
+                // Using adaptive rate sampling
+                builder.Use(x =>
+                    new AdaptiveSamplingTelemetryProcessor(samplingOptions.Adaptive, null, x)
```

Comment:
Does it seem that not all SamplingOptions properties are used here?

```
"InitialSamplingPercentage": "100",
                    "MinSamplingPercentage": "0.1",
                    "MaxSamplingPercentage": "100",
                    "EvaluationInterval": "00:00:15",
                    "SamplingPercentageDecreaseTimeout": "00:02:00",
                    "SamplingPercentageIncreaseTimeout": "00:15:00",
```

Reply:
Used all the options for adaptive.